packages/terra-functional-testing/src/terra-cli/wdio/test-runner.js

```diff
@@ -29,39 +31,69 @@ class TestRunner {
+   * @param {string} options.browsers - A list of browsers for the test run.
    * @param {string} options.config - A file path to the test runner configuration.
+   * @param {boolean} options.disableSeleniumService - A flag to disable the selenium docker service.
+   * @param {string} options.externalHost - The host address the testing environment is connected to.
+   * @param {number} options.externalPort - The port mapping from the host to the container.
    * @param {string} options.formFactor - A form factor for the test run.
-   * @param {string} options.locale - A language locale for the test run.
-   * @param {string} options.theme - A theme for the test run.
+   * @param {string} options.gridUrl - The remote selenium grid address.
    * @param {string} options.hostname - Automation driver host address.
+   * @param {boolean} options.keepAliveSeleniumDockerService - Determines to keep the selenium docker service running upon test completion.
+   * @param {string} options.locale - A language locale for the test run.
    * @param {number} options.port - Automation driver port.
-   * @param {string} options.baseUrl - The base url.
-   * @param {array} options.suite - Overrides specs and runs only the defined suites.
+   * @param {string} options.site - A file path to a static directory of assets. When defined, an express server will launch to serve the assets and disable running webpack..
    * @param {array} options.spec - A list of spec file paths.
-   * @param {Object} options.launcherOptions - Custom configuration launcher options. i.e. keepAliveSeleniumDockerService; updateScreenshots.
+   * @param {array} options.suite - Overrides specs and runs only the defined suites.
+   * @param {string} options.theme - A theme for the test run.
+   * @param {boolean} options.updateScreenshots - Updates all reference screenshots with the latest screenshots.
    * @returns {Promise} A promise that resolves with the test run exit code.
    */
   static async run(options) {
     let exitCode;
 
     try {
       const {
+        assetServerPort,
+        baseUrl,
+        browsers,
         config,
+        disableSeleniumService,
+        externalHost,
+        externalPort,
         formFactor,
+        gridUrl,
+        hostname,
+        keepAliveSeleniumDockerService,
         locale,
+        site,
         theme,
-        ...additionalLauncherOptions // hostname, port, baseUrl, suite, spec, launcherOptions
+        updateScreenshots,
+        ...cliOptions // spec, suite,
       } = options;
 
-      process.env.LOCALE = locale;
-      process.env.THEME = theme;
-
-      if (formFactor) {
-        process.env.FORM_FACTOR = formFactor;
-      }
+      const launcherOptions = {
+        ...cliOptions,
+        baseUrl: `http://${externalHost || getIpAddress()}:${externalPort || 8080}`,
+        capabilities: getCapabilities(browsers, !!gridUrl),
+        hostname: gridUrl || hostname || 'localhost',
+        port: gridUrl ? 80 : 4444,
```

Comment:
I want to discuss in our next sync on whether we set the following CLI options as is provided thru the runner or keep these logics in place. If we use these as they are provided, we are giving the responsibilities to the consumers to set these correctly. This goes back to @StephenEsser's comment [here](https://github.com/cerner/terra-toolkit/pull/582#issue-577197761) on whether to keep them.

- baseUrl
- port
- hostname

Reply:
My vote is to remove these options from the cli. As implemented today I don't think all the necessary configurations are in sync. Specifically regarding the port mapping in the docker-compose file. We could add these options back in the future if we encounter a scenario we need to support.

Reply:
I also vote to remove

Reply:
Removed baseUrl, hostname, and port CLI options: https://github.com/cerner/terra-toolkit/pull/583/commits/70319fe7996b688f517f3cbaf0d2eb7d3947111b